IDE/iotsafe/devices.c

```diff
@@ -0,0 +1,467 @@
+/* devices.c
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+/* Minimalist BSP for IoT-Safe example based on
+ * ST P-L596G-CELL02 + Quectel BG96 modem
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "devices.h"
+#define RTSCTS 0
+#define AUTOBR 0
+#define NVIC_UART1_IRQN          (37)
+#define NVIC_UART2_IRQN          (38)
+
+char modem_rx_buf[256];
+uint32_t modem_rx_idx = 0;
+uint32_t modem_parser_idx = 0;
+
+static void usart1_init(void)
+{
+    uint32_t reg;
+
+    /* Enable PWR */
+    RCC_APB1_ENR |= PWR_APB1_CLOCK_ER_VAL;
+    /* Enable GPIOG */
+    RCC_AHB2_ENR |= GPIOG_AHB2_CLOCK_ER_VAL;
+
+
+    /* Enable VDDIO2 */
+    while ((PWR_CR2 & PWR_CR2_IOSV) == 0) {
+        PWR_CR2 |= PWR_CR2_IOSV;
+        printf("Turning on VDDIO2\n");
+        sleep_ms(1000);
+    }
+    /* Enable GPIOB */
+    RCC_AHB2_ENR |= GPIOB_AHB2_CLOCK_ER_VAL;
+    /* Set mode = AF */
+    reg = GPIO_MODE(GPIOG_BASE) & ~(0x03 << (USART1_PIN_RX * 2));
+    GPIO_MODE(GPIOG_BASE) = reg | (0x02 << (USART1_PIN_RX * 2));
+
+    reg = GPIO_MODE(GPIOB_BASE) & ~(0x03 << (USART1_PIN_TX * 2));
+    GPIO_MODE(GPIOB_BASE) = reg | (0x02 << (USART1_PIN_TX * 2));
+
+    reg = GPIO_PUPD(GPIOG_BASE) & (0x03 << (USART1_PIN_RX * 2));
+    //GPIO_PUPD(GPIOG_BASE) = reg | (0x01 << (USART1_PIN_RX * 2));
```

Comment:
Should this line be removed or a comment added? Please use `/* */` style comments.